### ISSUE TITLE
str.fomat index_string too restricted

### ISSUE THREAD

**Issue by pmhahn** [python/mypy#10083]
**Bug Report**

The following code is flagged by `mypy-0.800`:
```python
"{0[k-y]}".format({"k-y": "val"})
```
The error is
> Invalid index expression in format field accessor "[k-y]"

According to [Python Format String Syntax](https://docs.python.org/3/library/string.html#formatstrings):
> replacement_field ::=  "{" [field_name] …
> field_name        ::=  arg_name ( … | "[" element_index "]")*
> element_index     ::=  … | index_string
> index_string      ::=  <any source character except "]"> +

the `-` should be okay. Similar when using `+` or `/`. Looks like only `[0-9A-Z_a-z]` are allowed with `mypy`.

**To Reproduce**

```
mypy -c '"{0[k-y]}".format({"k-y": ""})'
<string>:1: error: Invalid index expression in format field accessor "[k-y]"
```

**Expected Behavior**

No error should be detected.

**Actual Behavior**

`mypy` prints an error.

**Your Environment**

Debian-10-Buster with self-built  `mypy-0.800`
